File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    const arrMyString = my_string.split("");

    queries.forEach(([start, end]) => {
        // end + 1을 해서 end 인덱스까지 포함
        const arrReverse = arrMyString.slice(start, end + 1).reverse();

        // 교체할 길이는 (end - start + 1)
        arrMyString.splice(start, end - start + 1, ...arrReverse);
    });

    return arrMyString.join("");
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        char[] arr = my_string.toCharArray();
        for(int i=0; i<queries.length; i++) {
            int[] query = queries[i];
            reverse(arr, query);
        }
        return new String(arr);
    }

    private void reverse(char[] arr, int[] query) {
        int left = query[0];
        int right = query[1];
        while(left < right) {
            char c = arr[left];
            arr[left] = arr[right];
            arr[right] = c;
            left++;
            right--;
        }
    }
}